docs/site/Express-middleware.md

```diff
@@ -0,0 +1,327 @@
+  async handle(context: RequestContext): Promise<void> {
+    try {
+      const {request, response} = context;
+      // `this.invokeMiddleware` is an injected function to invoke a list of
+      // Express middleware handler functions
+      const finished = await this.invokeMiddleware(context, middlewareList);
+      if (finished) {
+        // The http response has already been produced by one of the Express
+        // middleware. We should not call further actions.
+        return;
+      }
+      const route = this.findRoute(request);
+      const args = await this.parseParams(request, route);
+      const result = await this.invoke(route, args);
+
+      this.send(response, result);
+    } catch (error) {
+      this.reject(context, error);
+    }
+  }
+}
+```
+
+#### Register Express middleware to be executed by `InvokeMiddleware` actions
+
+While the explicit Express middleware invocation is easy and simple, there are
+some limitations.
+
+1. The list of Express middleware has to be hard-coded in `src/sequence.ts`.
+   It's not easy to plug in a new middleware.
+2. The configuration of Express middleware is fixed unless we use dependency
+   injections for such values in `src/sequence.ts`.
+
+We provide another option to make invocation of Express middleware more flexible
+and extensible. The `InvokeMiddleware` actions within the sequence can discover
+registered middleware and invoke them in a chain.
+
+We first register middleware against the default or a named chain using APIs
+from `RestApplication`. It can happen in the constructor of an application.
+
+{% include code-caption.html content="src/application.ts" %}
+
+```ts
+import morgan from 'morgan';
+import {ApplicationConfig} from '@loopback/core';
+import {RestApplication} from '@loopback/rest';
+
+export class MyApplication extends RestApplication {
+  constructor(config: ApplicationConfig) {
+    this.expressMiddleware(
+      morgan,
+      {}, // default config
+      {
+        // Allow configuration to be injected to allow dynamic changes to
+        // morgan logging by configuring `middleware.morgan` to a new value
+        injectConfiguration: true,
+        key: 'middleware.morgan',
+      },
+    );
+  }
+}
+```
+
+### Use Express middleware as interceptors for controllers
+
+The LoopBack global and local interceptors now also serve as an avenue to attach
+middleware logic to specific points of controller invocations, such as global,
+class, or method levels.
+
+We create a few helper functions to wrap an Express middleware module into an
+LoopBack 4 interceptor.
+
+#### Adapt an Express middleware handler function to an interceptor
+
+```ts
+import {toInterceptor} from '@loopback/express';
+import morgan from 'morgan';
+
+const morganInterceptor = toInterceptor(morgan('combined'));
+```
+
+#### Create an interceptor from Express middleware factory function
+
+```ts
+import {createInterceptor} from '@loopback/express';
```

Comment:
could you explain more about when use `createInterceptor` and when use `toInterceptor`?
e.g. for this example, why not using:

`const helmetInterceptor = toInterceptor(helmet(helmetConfig));`

Reply:
`toInterceptor()` takes the Express handler function directly while `createInterceptor` takes the factory function and config object.